pkg/apiserver/authentication/identityprovider/ldap/ldap.go

```diff
@@ -126,7 +126,10 @@ func (l ldapProvider) Authenticate(username string, password string) (identitypr
 		return nil, err
 	}
 
-	filter := fmt.Sprintf("(&(%s=%s)%s)", l.LoginAttribute, username, l.UserSearchFilter)
+	filter := fmt.Sprintf("(%s=%s)", l.LoginAttribute, ldap.EscapeFilter(username))
```

Comment:
Not sure whether this is releated with the feature. Please create another PR if it related some other issue..

Reply:
Use a clearer way to splice, code optimization.